lib/crapto/english_utils.nit

```diff
@@ -0,0 +1,68 @@
+# This file is part of NIT ( http://www.nitlanguage.org ).
+#
+# Copyright 2016 Philippe Pépos Petitclerc <ppeposp@gmail.com>
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+redef class Text
+
+	# Score `self` according to english's letter frequency.
+	# This function is useful mainly for cryptography but could happen to be helpful
+	# elsewhere.
+	#
+	#     assert "aaaa".english_scoring > "bbbb".english_scoring
+	fun english_scoring: Float do
+
+		var score = 0.0
+		var freqs = new HashMap[Char, Float]
```

Comment:
You should cache this Map in a lazy attribute of Sys so you don't regenerate it on each call to this method.